### PR TITLE
only acccept cloud config files for certain providers

### PR DESCRIPTION
**Pull request by deads2k**
/hold

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
/test verify

Not sure I get how this caused that error...

Change looks good otherwise and is needed to unblock work.

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 1]
```diff
@@ -92,8 +93,10 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 		Namespace: sourceCloudConfigNamespace,
 		Name:      sourceCloudConfigMap,
 	}
-	// we set cloudprovider configmap values only for vsphere.
-	if cloudProvider != "vsphere" {
+
+	// we set cloudprovider configmap values only for some cloud providers.
+	validCloudProviders := sets.NewString("azure", "vsphere")
+	if !validCloudProviders.Has(cloudProvider){
```
Comment (tnozicka) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: there should be a space at the end before `{`, right?

```
2019/05/03 01:01:46 Executing test verify
Running `gofmt -s -l` on 172 file(s).
./pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go
verify-gofmt failed - please run `make update-gofmt`
make: *** [verify-gofmt] Error 1
```

File: pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package cloudprovider
 
 import (
 	"fmt"
+	"k8s.io/apimachinery/pkg/util/sets"
```
Comment (tnozicka) [inline pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go]: since you will be fixing for gofmt, you might just move this one line down

**Pull request closed without merging**